front_end/webserver.py

```diff
@@ -709,7 +710,17 @@ def get(self, course, assignment):
                 curr_datetime = datetime.datetime.now()
                 start_time = content.get_user_assignment_start_time(course, assignment, user_info["user_id"])
 
-                if assignment_details["start_date"] and assignment_details["start_date"] > curr_datetime:
+                client_ip = self.request.headers.get("X-Real-IP") or \
```

Comment:
Thanks for finding these. There's another place in the code where we use `self.request.remote_ip`. Will you please create a function called `get_client_ip_address` that checks all the ways you are checking here and then returns the client's IP address? Then in that other place, invoke this new function as well.